src/main/java/org/primefaces/component/datatable/DataTableRenderer.java

```diff
@@ -842,122 +842,151 @@ else if (headerText != null) {
+            }
 
-                writer.startElement("select", null);
-                writer.writeAttribute("id", filterId, null);
-                writer.writeAttribute("name", filterId, null);
-                writer.writeAttribute("class", filterStyleClass, null);
-                writer.writeAttribute(HTML.ARIA_LABELLEDBY, ariaLabelId, null);
+            writer.endElement("option");
+        }
 
-                if (disableTabbing) {
-                    writer.writeAttribute("tabindex", "-1", null);
-                }
+        writer.endElement("select");
+    }
 
-                SelectItem[] itemsArray = getFilterOptions(column);
+    protected void encodeFilterInputText(UIColumn column, ResponseWriter writer, boolean disableTabbing,
+        String filterId, String filterStyleClass, Object filterValue, String ariaLabelId) throws IOException {
 
-                for (SelectItem item : itemsArray) {
-                    Object itemValue = item.getValue();
+        filterStyleClass = filterStyleClass == null
+                           ? DataTable.COLUMN_INPUT_FILTER_CLASS
+                           : DataTable.COLUMN_INPUT_FILTER_CLASS + " " + filterStyleClass;
 
-                    writer.startElement("option", null);
-                    writer.writeAttribute("value", item.getValue(), null);
-                    if (itemValue != null && String.valueOf(itemValue).equals(filterValue)) {
-                        writer.writeAttribute("selected", "selected", null);
-                    }
+        writer.startElement("input", null);
+        writer.writeAttribute("id", filterId, null);
+        writer.writeAttribute("name", filterId, null);
+        writer.writeAttribute("class", filterStyleClass, null);
+        writer.writeAttribute("value", filterValue, null);
+        writer.writeAttribute("autocomplete", "off", null);
+        writer.writeAttribute(HTML.ARIA_LABELLEDBY, ariaLabelId, null);
 
-                    if (item.isEscape()) {
-                        writer.writeText(item.getLabel(), "value");
-                    }
-                    else {
-                        writer.write(item.getLabel());
-                    }
+        if (disableTabbing) {
+            writer.writeAttribute("tabindex", "-1", null);
+        }
 
-                    writer.endElement("option");
-                }
+        if (column.getFilterStyle() != null) {
+            writer.writeAttribute("style", column.getFilterStyle(), null);
+        }
 
-                writer.endElement("select");
-            }
+        if (column.getFilterMaxLength() != Integer.MAX_VALUE) {
+            writer.writeAttribute("maxlength", column.getFilterMaxLength(), null);
+        }
+
+        writer.endElement("input");
+    }
+
+    protected Object findFilterValueForColumn(FacesContext context, DataTable table,
+        UIColumn column, String filterId) {
+
+        Object filterValue;
+        if (table.isReset()) {
+            filterValue = "";
```

Comment:
Use Constants.EMPTY_STRING instead of "".

Reply:
Well, I just did refactoring, no changes to code otherwise. But I will do if you want me to.